app/Repositories/Three/SignupRepository.php

```diff
@@ -10,26 +10,27 @@ class SignupRepository
      * Create a signup.
      *
      * @param  array $data
+     * @param  string $northstarIdFromAuth
      * @return \Rogue\Models\Signup|null
      */
-    public function create($data)
+    public function create($data, $northstarIdFromAuth = null)
     {
         // Create the signup
         $signup = new Signup;
 
-        $signup->northstar_id = $data['northstar_id'];
+        $signup->northstar_id = isset($northstarIdFromAuth) ? $northstarIdFromAuth : $data['northstar_id'];
```

Comment:
Updated this based on @DFurnes 's suggestion: 

>  It might make sense to send these along as parameters from the controller (so that we could use the repository for things that aren't HTTP requests, like for creating signups from a CSV or something).

The only thing I'm worried about here is if somehow, the `auth()->id()` isn't sent and `$data['northstar_id']` also isn't set, this will break? Because we don't have a validation that requires to send if the `northstar_id` in the request anymore and all signups need to be associated to a user.

Reply:
Yeah I think we'd always want to provide the ID as a parameter (so even if we were creating from a CSV or an admin flow or something, we'd still call `$repository->create($data, $id)` with the proper ID).